### PR TITLE
Bulk Download Files

### PR DESCRIPTION
**Pull request by utku-ozturk**
1. Custom FileSearchView added to enable bulk download of files.
2. BrowseView generated metadata.tsv is updated to reflect new columns.

Screen_Shot_2020-04-01_at_13 14 38

### CODE REVIEW

File: src/encoded/batch_download.py [hunk 1]
```diff
@@ -469,14 +485,17 @@ def format_file(f, exp, exp_row_vals, exp_set, exp_set_row_vals):
...
+                else:
+                    xfile_vals['Size (MB)'] = None
                 xfile_vals['Related File Relationship'] = 'secondary file for'
                 xfile_vals['Related File'] = all_row_vals.get('File Accession')
                 files_returned.append(xfile_vals)
 
         return files_returned
 
-
+
```
Comment (j1z0) [inline src/encoded/batch_download.py:498]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/encoded/batch_download.py  2
         

Clones added
============
- src/encoded/static/components/browse/FileSearchView.js  12
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993&bid=17442288)

File: src/encoded/batch_download.py [hunk 2]
```diff
@@ -68,13 +73,23 @@ def includeme(config):
...
+    ('Experiment Type',             (FILE_ONLY, ['track_and_facet_info.experiment_type'])),
+    ('Replicate Info',              (FILE_ONLY, ['track_and_facet_info.replicate_info'])),
+    ('Assay Details',               (FILE_ONLY, ['track_and_facet_info.assay_info'])),
+    ('Biosource',                   (FILE_ONLY, ['track_and_facet_info.biosource_name'])),
+    ('Condition ',                  (FILE_ONLY, ['track_and_facet_info.condition'])), #do not remove trailing whitespace
+    ('Dataset ',                    (FILE_ONLY, ['track_and_facet_info.dataset'])), #do not remove trailing whitespace
+    ('In Experiment As',            (FILE_ONLY, ['track_and_facet_info.experiment_bucket'])),
+    ('Project',                     (EXP_SET,   ['award.project'])),
+    ('Generating Lab',              (FILE,      ['lab.display_title'])),
+    ('Contributing Lab',            (FILE,      ['contributing_labs.display_title'])),
```
Comment (j1z0) [inline src/encoded/batch_download.py:89]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py [hunk 3]
```diff
@@ -68,13 +73,23 @@ def includeme(config):
    #('Digestion Enzyme',            (EXP,       ['digestion_enzyme.name'])),
     ('Related File Relationship',   (FILE,      ['related_files.relationship_type'])),
     ('Related File',                (FILE,      ['related_files.file.accession'])),
-    ('Paired end',                  (FILE,      ['paired_end'])),
-    ('Lab',                         (EXP_SET,   ['lab.display_title'])),
-    ('Project',                     (EXP_SET,   ['award.project'])),
+    ('Paired End',                  (FILE,      ['paired_end'])),
```
Comment (j1z0) [inline src/encoded/batch_download.py:76]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/encoded/batch_download.py  2
         

Clones added
============
- src/encoded/static/components/browse/FileSearchView.js  12
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993&bid=17442288)

File: src/encoded/batch_download.py (hunk 2 above)
Comment (j1z0) [inline src/encoded/batch_download.py:89]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py [hunk 4]
```diff
@@ -358,10 +373,11 @@ def should_file_row_object_be_included(column_vals_dict):
         '''Ensure row's ExpSet, Exp, and File accession are in list of accession triples sent in URL params.'''
         if accession_triples is None:
             return True
+
```
Comment (j1z0) [inline src/encoded/batch_download.py:376]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 3 above)
Comment (j1z0) [inline src/encoded/batch_download.py:76]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

**Comment by utku-ozturk** [comment]
Re: Notes subtitle
@Bianca-Morris  Good point! If any unpublished file found, there is a bullet point under Notes subtitle. I think, we all tested to dowload files at least having one unpublished file. I'll commit a minor fix shortly! Thanks.

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/encoded/batch_download.py  2
         

Clones added
============
- src/encoded/static/components/browse/FileSearchView.js  12
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993&bid=17442288)

File: src/encoded/batch_download.py (hunk 2 above)
Comment (j1z0) [inline src/encoded/batch_download.py:89]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 4 above)
Comment (j1z0) [inline src/encoded/batch_download.py:376]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 3 above)
Comment (j1z0) [inline src/encoded/batch_download.py:76]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/encoded/batch_download.py  2
         

Clones added
============
- src/encoded/static/components/browse/FileSearchView.js  12
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993&bid=17442288)

File: src/encoded/batch_download.py (hunk 2 above)
Comment (j1z0) [inline src/encoded/batch_download.py:89]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 4 above)
Comment (j1z0) [inline src/encoded/batch_download.py:376]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 3 above)
Comment (j1z0) [inline src/encoded/batch_download.py:76]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

**Comment by utku-ozturk** [comment]
> Replace "which are not yet available" with "which are currently not available"

@andreacosolo text replaced.

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/encoded/batch_download.py  2
         

Clones added
============
- src/encoded/static/components/browse/FileSearchView.js  12
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993&bid=17442288)

File: src/encoded/batch_download.py (hunk 2 above)
Comment (j1z0) [inline src/encoded/batch_download.py:89]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 4 above)
Comment (j1z0) [inline src/encoded/batch_download.py:376]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 3 above)
Comment (j1z0) [inline src/encoded/batch_download.py:76]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/encoded/batch_download.py  2
         

Clones added
============
- src/encoded/static/components/browse/FileSearchView.js  12
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993&bid=17442288)

File: src/encoded/batch_download.py (hunk 2 above)
Comment (j1z0) [inline src/encoded/batch_download.py:89]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 4 above)
Comment (j1z0) [inline src/encoded/batch_download.py:376]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 3 above)
Comment (j1z0) [inline src/encoded/batch_download.py:76]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py [hunk 5]
```diff
@@ -465,18 +479,19 @@ def format_file(f, exp, exp_row_vals, exp_set, exp_set_row_vals):
...
         if f.get('extra_files') and len(f['extra_files']) > 0:
             for xfile in f['extra_files']:
+                if xfile.get('use_for') == 'visualization':
+                    continue
                 xfile_vals = all_row_vals.copy()
                 xfile_vals['File Download URL'] = request.host_url + xfile['href'] if xfile.get('href') else None
                 xfile_vals['File Format'] = xfile.get('file_format', {}).get('display_title')
                 xfile_vals['md5sum'] = xfile.get('md5sum')
-                xfile_vals['Size'] = xfile.get('file_size')
+                xfile_vals['Size (MB)'] = xfile.get('file_size')
```
[the remark this answers is not in the document]
Comment (utku-ozturk) [reply src/encoded/batch_download.py:488]: @alexkb0009 yes, the actual file size is in bytes here. On the other hand, if we divide 1024 * 1024 here, then the value will also be divided in `post_process_file_row_dict(file_row_dict_tuple)` function once again, so we get error. (Very first implementation had double division, but removed.)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/encoded/batch_download.py  2
         

Clones added
============
- src/encoded/static/components/browse/FileSearchView.js  12
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993&bid=17442288)

File: src/encoded/batch_download.py (hunk 2 above)
Comment (j1z0) [inline src/encoded/batch_download.py:89]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 4 above)
Comment (j1z0) [inline src/encoded/batch_download.py:376]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 3 above)
Comment (j1z0) [inline src/encoded/batch_download.py:76]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- src/encoded/batch_download.py  2
         

Clones added
============
- src/encoded/static/components/browse/FileSearchView.js  12
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993&bid=17442288)

File: src/encoded/batch_download.py (hunk 2 above)
Comment (j1z0) [inline src/encoded/batch_download.py:89]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 4 above)
Comment (j1z0) [inline src/encoded/batch_download.py:376]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 3 above)
Comment (j1z0) [inline src/encoded/batch_download.py:76]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/search.py [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import re
 import math
 import itertools
+from functools import reduce
```
Comment (j1z0) [inline src/encoded/search.py:4]: Codacy Issue found: [Unused reduce imported from functools](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- src/encoded/batch_download.py  2
         

Clones added
============
- src/encoded/static/components/browse/FileSearchView.js  12
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993&bid=17442288)

File: src/encoded/batch_download.py (hunk 2 above)
Comment (j1z0) [inline src/encoded/batch_download.py:89]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 4 above)
Comment (j1z0) [inline src/encoded/batch_download.py:376]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 3 above)
Comment (j1z0) [inline src/encoded/batch_download.py:76]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/search.py (hunk 6 above)
Comment (j1z0) [inline src/encoded/search.py:4]: Codacy Issue found: [Unused reduce imported from functools](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/search.py [hunk 7]
```diff
@@ -262,36 +262,51 @@ def collection_view(context, request):
...
+    :param doc_types: Type names we would like to search on.
     :return: search_types, or a list of 'SearchResults' type candidates
     """
-    search_types = []
-    if len(doc_types) == 1:  # if we have one, add it and its base_type
-        ti = types[doc_types[0]]
-        search_types.append(ti.name + "SearchResults")
-        if hasattr(ti, 'base_types'):
+    encompassing_ti_for_all_items = None
+    if len(doc_types) < 1:
```
Comment (alexkb0009) [inline src/encoded/search.py:273]: Actually I don't think this is true, was another glitch (forgot to set `encompassing_ti_for_all_items = ti` under `if encompassing_ti_for_all_items is None` .. will delete + test.

File: src/encoded/batch_download.py [hunk 8]
```diff
@@ -41,25 +41,30 @@ def includeme(config):
...
+FILE        = 2
+FILE_ONLY   = 3
 
 # includes concatenated properties
 TSV_MAPPING = OrderedDict([
     ('File Download URL',           (FILE,      ['href'])),
     ('Experiment Set Accession',    (EXP_SET,   ['accession'])),
     ('Experiment Accession',        (EXP,       ['accession'])),
+    ('Experiment Set Accession ',   (FILE_ONLY, ['experiment_sets.accession'])), #do not remove trailing whitespace
+    ('Experiment Accession ',       (FILE_ONLY, ['experiments.accession'])), #do not remove trailing whitespace
```
Comment (alexkb0009) [inline src/encoded/batch_download.py:55]: Haha

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- src/encoded/batch_download.py  2
         

Clones added
============
- src/encoded/static/components/browse/FileSearchView.js  12
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993&bid=17442288)

File: src/encoded/search.py (hunk 6 above)
Comment (j1z0) [inline src/encoded/search.py:4]: Codacy Issue found: [Unused reduce imported from functools](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 2 above)
Comment (j1z0) [inline src/encoded/batch_download.py:89]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 4 above)
Comment (j1z0) [inline src/encoded/batch_download.py:376]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 3 above)
Comment (j1z0) [inline src/encoded/batch_download.py:76]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- src/encoded/batch_download.py  2
         

Clones added
============
- src/encoded/static/components/browse/FileSearchView.js  12
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993&bid=17442288)

File: src/encoded/batch_download.py (hunk 2 above)
Comment (j1z0) [inline src/encoded/batch_download.py:89]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 4 above)
Comment (j1z0) [inline src/encoded/batch_download.py:376]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 3 above)
Comment (j1z0) [inline src/encoded/batch_download.py:76]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/search.py (hunk 6 above)
Comment (j1z0) [inline src/encoded/search.py:4]: Codacy Issue found: [Unused reduce imported from functools](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: package.json [hunk 9]
```diff
@@ -100,14 +100,14 @@
...
-    "@hms-dbmi-bgm/shared-portal-components": "github:4dn-dcic/shared-portal-components#0.0.2.61.3",
+    "@hms-dbmi-bgm/shared-portal-components": "github:4dn-dcic/shared-portal-components#0.0.2.62.3b",
     "4dn-microscopy-metadata-tool": "github:WU-BIMAC/4DNMicroscopyMetadataToolReact#0.0.0.18",
     "babel-polyfill": "^6.26.0",
     "d3": "^5.12.0",
     "detect-browser": "^3.0.1",
     "domready": "^0.3.0",
     "form-serialize": "^0.6.0",
-    "higlass": "^1.6.11",
+    "higlass": "^1.9.3",
```
Comment (alexkb0009) [inline package.json:110]: Noticed this was already being updated to 1.9.x in package-lock.json so bumped it up here to match.

File: src/encoded/static/components/browse/BrowseView.js [hunk 10]
```diff
@@ -569,7 +563,7 @@ function BrowseTableWithSelectedFilesCheckboxes(props){
     return (
         <WindowNavigationController {...{ href, context }} navigate={propNavigate}>
             <ColumnCombiner columnExtensionMap={columnExtensionMapWithSelectedFilesCheckboxes}>
-                <CustomColumnController>
+                <CustomColumnController {...{ windowWidth }}>
```
Comment (alexkb0009) [inline src/encoded/static/components/browse/BrowseView.js:566]: Might not be needed here... eh.. will remove.

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- src/encoded/batch_download.py  2
         

Clones added
============
- src/encoded/static/components/browse/BrowseView.js  6
- src/encoded/static/components/browse/FileSearchView.js  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993&bid=17442288)

File: src/encoded/batch_download.py (hunk 2 above)
Comment (j1z0) [inline src/encoded/batch_download.py:89]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 4 above)
Comment (j1z0) [inline src/encoded/batch_download.py:376]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/search.py (hunk 6 above)
Comment (j1z0) [inline src/encoded/search.py:4]: Codacy Issue found: [Unused reduce imported from functools](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 3 above)
Comment (j1z0) [inline src/encoded/batch_download.py:76]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- src/encoded/batch_download.py  2
         

Clones added
============
- src/encoded/static/components/browse/BrowseView.js  6
- src/encoded/static/components/browse/FileSearchView.js  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993&bid=17442288)

File: src/encoded/batch_download.py (hunk 4 above)
Comment (j1z0) [inline src/encoded/batch_download.py:376]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/batch_download.py (hunk 3 above)
Comment (j1z0) [inline src/encoded/batch_download.py:76]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

File: src/encoded/search.py (hunk 6 above)
Comment (j1z0) [inline src/encoded/search.py:4]: Codacy Issue found: [Unused reduce imported from functools](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5328993)

**Pull request merged**